compiler+runtime/src/jank/clojure/core.jank

```diff
@@ -149,6 +150,37 @@
 (def seqable?
   "Return true if the seq function is supported for x"
   clojure.core-native/seqable?)
+
+(def ^:private reverse*
+  (fn* reverse* [coll]
+    (loop* [acc '()
+            coll coll]
```

Comment:
IMO should be initialized to `(seq coll)` so we're not reasoning about a mutable collection.

Reply:
What would that change?

Reply:
There's a couple of subtle possibilities.

Some collections like `eduction` don't cache their values at all. So testing `empty?` on an eduction doesn't necessarily mean it has a `first`. Both calls might, say, hit the network to rebuild the seq.

And there's a race condition for mutable collections. A mutable collection could change its size between the `empty?` call and the `first` call. If it changes to empty, then the output will be `(cons nil ...)`.

Reply:
I think the first case is far-fetched, given what jank supports, and the second case is a user error. But I see your argument more as a common principle for iteration, rather than for this specific *private* var. So, taking the argument that way, we should be consistent in iterating over seqs to avoid these classes of problems. That's fair.

Reply:
Yes. I think of it in terms of the input type of `reverse`, which is `seqable`. All it does is return a `seq`, with no guarantees that it's cached or idempotent.